### PR TITLE
New action details UI

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

See https://github.com/dust-tt/dust/issues/5350.

This PR establishes the groundwork for the new Action details rendering and implements the [design](https://www.figma.com/design/QOxHwppG64GtPocpDhS6Y7/Product?node-id=7026-16895&m=dev) for the retrieval action.

_It exposes the existing issue with our action type (please overlook the temporary fixes until we address these in a refactor 
 -- which is my next task)._

The approach introduced in this PR aims to standardize the design across all our actions and ensure type safety consistently. While the design post this PR isn't perfect, the incremental refactoring should be okay, especially since Multi action is currently only relevant for Dust and a few others.

**It looks like this:**

image

### CODE REVIEW

File: front/components/assistant/conversation/RetrievalAction.tsx [hunk 1]
```diff
@@ -15,6 +15,7 @@ import type {
 import { Transition } from "@headlessui/react";
 import { useState } from "react";
 
+import { makeLinkForRetrievedDocument } from "@app/components/actions/retrieval/utils";
```
Comment (flvndvd) [inline front/components/assistant/conversation/RetrievalAction.tsx:18]: Will clean up this file in a follow up PR once all actions have been refactored.

File: front/components/actions/retrieval/utils.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
+import type { ConnectorProvider, RetrievalDocumentType } from "@dust-tt/types";
+
+type ConnectorProviderDocumentType =
+  | Exclude<ConnectorProvider, "webcrawler">
+  | "document";
+
+const providerMap: Record<string, ConnectorProviderDocumentType> = {
```
Comment (PopDaph) [inline front/components/actions/retrieval/utils.ts:7]: Can we factorize this that is also used in `providerFromDocument`?

File: front/components/actions/retrieval/utils.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+): ConnectorProviderDocumentType {
+  const match = document.dataSourceId.match(providerRegex);
+  if (match && match[1]) {
+    return providerMap[match[1]];
+  }
+
+  return "document";
+}
+
+function getTitleFromRetrievedDocument(
```
Comment (PopDaph) [inline front/components/actions/retrieval/utils.ts:29]: Actually isn't the logic from this doc already defined in "@app/components/assistant/conversation/RetrievalAction"?

Comment (flvndvd) [reply front/components/actions/retrieval/utils.ts:29]: Yeah I left a comment in the top of the legacy file. Today the logic is spread everywhere through the codebase. So the plan here is to first introduce new action details design before getting rid of all the legacy code.

**Review by PopDaph — APPROVED** [review]

**Pull request merged**